src/Request.php

```diff
@@ -132,7 +132,13 @@ public function handleData($data)
         $this->buffer .= $data;
 
         if (false !== strpos($this->buffer, "\r\n\r\n")) {
-            list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            try {
+                list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            } catch (\InvalidArgumentException $exception) {
+                $this->emit('error', [$exception, $this]);
+
+                return;
```

Comment:
Shouldn't we also unsubscribe from the stream as we do here https://github.com/djagya/http-client/blob/61bf248481da1101d432a57a9c2748a8af7016e0/src/Request.php#L145 since the response parsing has failed and nothing new coming from the stream could change that.

Ping @clue